arkade/src/commonMain/kotlin/com/arkade/core/batches/TxTree.kt

```
@@ -0,0 +1,109 @@
+        children.forEach { (_, child) ->
+            leaves.addAll(child.leaves())
+        }
+        return leaves
+    }
+
+    override fun iterator(): Iterator<TxTree> {
+        return this.iterator()
+        for (child in children.values) {
+            for (node in child) {
+                return node.iterator()
+            }
```

detekt (1.23.8)

[warning] This loop contains an unconditional jump expression which essentially renders it useless as it will exit the loop during the first iteration.

(detekt.potential-bugs.UnconditionalJumpStatementInLoop)